src/assets/wise5/components/match/match-student/match-student-default/match-student-default.component.ts

```diff
@@ -613,11 +613,17 @@ export class MatchStudentDefaultComponent extends ComponentStudent {
   }
 
   private addIdeasToSourceBucket(responses: any[], rubric: CRaterRubric): void {
-    getUniqueIdeas(responses, rubric).forEach((idea) => {
-      const choice = new Choice(idea.name, idea.text);
-      this.choices.push(choice);
-      this.getBucketById(this.sourceBucketId).items.push(choice);
-    });
+    getUniqueIdeas(responses, rubric)
+      .filter((idea) => !this.isInSourceBucket(idea.name))
+      .forEach((idea) => {
+        const choice = new Choice(idea.name, idea.text);
+        this.choices.push(choice);
+        this.getBucketById(this.sourceBucketId).items.push(choice);
+      });
+  }
+
+  private isInSourceBucket(choiceId: string): boolean {
+    return this.sourceBucket.items.some((item) => item.id === choiceId);
```

Comment:
```suggestion
      .filter((idea) => !this.isInSourceBucket(idea.text))
      .forEach((idea) => {
        const choice = new Choice(idea.name, idea.text);
        this.choices.push(choice);
        this.getBucketById(this.sourceBucketId).items.push(choice);
      });
  }

  private isInSourceBucket(choiceId: string): boolean {
    return this.sourceBucket.items.some((item) => item.value === choiceId);
```

Reply:
I believe we should be passing in `idea.text` instead of `idea.name` because we should be comparing the text description of the idea to the choice (`item.name` is just the idea number). We should also be checking against `item.value` which is the text content of the choice (not `item.id`, which is the randomly generated string identifier for the choice).